### PR TITLE
driver: spi: support it8xxx2 spi driver

### PR DESCRIPTION
**Pull request by Chenhongren**
This commit adds the it8xxx2 spi driver support.

Tested with:
- west build -p always -b it8xxx2_evb samples/drivers/spi_flash

### CODE REVIEW

File: drivers/spi/spi_it8xxx2.c [hunk 1]
```diff
@@ -0,0 +1,661 @@
...
+
+	LOG_DBG("chip select: %d, operation: 0x%x", spi_cfg->slave, spi_cfg->operation);
+
+	if (SPI_OP_MODE_GET(spi_cfg->operation) == SPI_OP_MODE_SLAVE) {
+		LOG_ERR("Unsupported SPI slave mode");
+		return -ENOTSUP;
+	}
+
+	if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_LOOP) {
+		LOG_ERR("Unsupported the loopback mode");
```
[the remark this answers is not in the document]
Comment (Chenhongren) [reply drivers/spi/spi_it8xxx2.c]: fixed

File: drivers/spi/spi_it8xxx2.c [hunk 2]
```diff
@@ -0,0 +1,661 @@
...
+		return -ENOTSUP;
+	}
+
+	if (SPI_MODE_GET(spi_cfg->operation) & SPI_MODE_CPHA) {
+		LOG_ERR("Unsupported the spi mode 1 and mode 3");
```
[the remark this answers is not in the document]
Comment (Chenhongren) [reply drivers/spi/spi_it8xxx2.c]: fixed

File: drivers/spi/spi_it8xxx2.c [hunk 3]
```diff
@@ -0,0 +1,661 @@
...
+	uint32_t operation = spi_cfg->operation;
+
+	switch (operation & SPI_LINES_MASK) {
+	case SPI_LINES_SINGLE:
+		spi_it8xxx2_enable_3_wire(dev, spi_cfg->slave, false);
+		spi_it8xxx2_enable_dual_pin(dev, false);
+		spi_it8xxx2_enable_quad_pin(dev, false);
+		LOG_DBG("Single mode is selected");
+		break;
+	case SPI_LINES_DUAL:
```
[the remark this answers is not in the document]
Comment (Chenhongren) [reply drivers/spi/spi_it8xxx2.c:228]: Did you mean the API for selecting input direction in dual/quad modes? (2 x I/O, 1I/2O, 4 x I/O, 1I/4O)

Currently, in it8xxx2 driver, 2 x I/O or 4 x I/O transfer mode are selected when the SPI_LINES_DUAL or SPI_LINES_QUAD bit is enabled. Unfortunately, we’re unable to switch to 1I/2O or 1I/4O mode as there doesn’t appear to be an API available for those configurations.

[the remark this answers is not in the document]
Comment (Chenhongren) [reply drivers/spi/spi_it8xxx2.c:228]: Noted with thanks. We'll remove support for dual and quad modes in this pr.
BTW, is there any plan to support these APIs in the future?

[the remark this answers is not in the document]
Comment (Chenhongren) [reply drivers/spi/spi_it8xxx2.c]: The change has been uploaded.

**Review by tbursztyka — APPROVED** [review]

**Review by Dino-Li — APPROVED** [review]

**Pull request merged**